### ISSUE TITLE
Add support for all Youtube Music shortcuts

### ISSUE THREAD

**Issue by PolarGoose** [th-ch/youtube-music#106]
The Shortcuts plugin doesn't support all shortcuts available in Youtube music web interface.
I have found a nice summary of all shortcuts: https://support.google.com/youtubemusic/thread/180145?hl=en

**Comment by th-ch** [comment]
Hey @PolarGoose, thanks for the feedback! The shortcuts plugin maps custom bindings to existing shortcuts (eg media keys or ctrl+F for search, etc) but all YTM shortcuts should be supported when the app is in the foreground - do you have an example of shortcut that does not work/did you have custom key bindings in mind?

**Comment by PolarGoose** [comment]
@th-ch,

I thought the the shorcuts plugin allows controlling YTM while it is in a background. Isn't it the case?

**Comment by th-ch** [comment]
Hey @PolarGoose, the shortcuts plugins actually adds additional shortcuts in addition to the default YTM ones:
- media keys (global shortcuts)
- some new combinations, like `ctrl/cmd + F` to search (local shortcuts, can only be used when YTM is in the foreground)

Some shortcuts might create issues if registered globally (for example, people might not want to mute/unmute YTM each time they press M) - Do you have a particular shortcut that you would like registered globally in mind?

A possible solution could be to add custom global/local shortcuts entries in the advanced config, this way each user can customize shortcuts like they want!

**Comment by PolarGoose** [comment]
@th-ch,
Exactly, I want is an ability to assign custom global hotkeys for every local youtube shortcut.

**Comment by th-ch** [comment]
Hey @PolarGoose, support for custom shortcuts has been added in https://github.com/th-ch/youtube-music/commit/024ed9085c9c8e3b58fdbfa26f781bc6d4ae2c56 and will ship in the next version! In the advanced options, you can add custom local and global shortcuts with the following format:
```json
"shortcuts": {
    "enabled": true,
    "local": [
        {
            "shortcut": "CommandOrControl+X",
            "action": "goToSettings"
        }
    ],
    "global": [
        {
            "shortcut": "CommandOrControl+Y",
            "action": "maximizeMinimisePlayer"
        }
    ]
},
```

Shortcut syntax in documented in [the Electron doc about accelerators](https://www.electronjs.org/docs/api/accelerator) and possible actions are listed in [registered actions](https://github.com/th-ch/youtube-music/blob/5a1358b3100c9de776b044494d940ee2a7e0146d/providers/song-controls.js#L13).

**Comment by PolarGoose** [comment]
@th-ch,
Thank you for the implementation.

**Comment by ActualBest** [comment]
@th-ch 
Can I disable shortcuts this way? Would like to use the shortcuts plugin without hijacking my media controls :)
Loving this app btw, thanks so much to u and other contributers

**Comment by ActualBest** [comment]
> @th-ch
> Can I disable shortcuts this way? Would like to use the shortcuts plugin without hijacking my media controls :)
> Loving this app btw, thanks so much to u and other contributers

Update: I found a way to disable specific included shortcuts with shortcuts plugin, will make a pr hopefully by the end of Sat

**Comment by ActualBest** [comment]
> > @th-ch
> > Can I disable shortcuts this way? Would like to use the shortcuts plugin without hijacking my media controls :)
> > Loving this app btw, thanks so much to u and other contributers
> 
> Update: I found a way to disable specific included shortcuts with shortcuts plugin, will make a pr hopefully by the end of Sat

Update update: it won't be needed once https://github.com/th-ch/youtube-music/pull/243/commits/964974c142837429d4003fca38c7a8c4fb237121 is merged

**Comment by Mohmadmiui** [comment]
Does anyone know how to make a shortcut to increase volume on the up arrow and decrease volume on the down arrow?

**Comment by Araxeus** [comment]
> Does anyone know how to make a shortcut to increase volume on the up arrow and decrease volume on the down arrow?

Precise volume plugin should be able to do that (when the window is focused)

**Comment by Mohmadmiui** [comment]
I need it when it unfocused. To regulate song volume when I'm in game and etc.
